sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -304,6 +310,18 @@ public String getId() {
      */
     public <T> Mono<CosmosItemResponse<T>> upsertItem(T item, CosmosItemRequestOptions options) {
         final CosmosItemRequestOptions requestOptions = options == null ? new CosmosItemRequestOptions() : options;
+
+        if (this.bulkExecutionEnabled) {
+            @SuppressWarnings("unchecked")
+            Class<T> itemType = (Class<T>) item.getClass();
+            return bulkExecutionInternal(ModelBridgeInternal.toRequestOptions(requestOptions),
+                item,
+                null,
```

Comment:
itemid intentionally hard-coded as null?

Reply:
Yeah. it is only needed in replace and delete. Should we do method overloading for bulkExecutionInternal?